src/main/java/net/snowflake/ingest/streaming/internal/SnowflakeStreamingIngestChannelInternal.java

```diff
@@ -302,30 +316,39 @@ CompletableFuture<Void> flush(boolean closing) {
   public CompletableFuture<Void> close() {
     checkValidation();
 
-    if (isClosed()) {
-      return CompletableFuture.completedFuture(null);
-    }
+    if (!markClosed()) {
+      this.owningClient.removeChannelIfSequencersMatch(this);
 
-    markClosed();
-    this.owningClient.removeChannelIfSequencersMatch(this);
-    return flush(true)
-        .thenRunAsync(
-            () -> {
-              List<SnowflakeStreamingIngestChannelInternal> uncommittedChannels =
-                  this.owningClient.verifyChannelsAreFullyCommitted(
-                      Collections.singletonList(this));
-
-              this.arrowBuffer.close();
-
-              // Throw an exception if the channel has any uncommitted rows
-              if (!uncommittedChannels.isEmpty()) {
-                throw new SFException(
-                    ErrorCode.CHANNEL_WITH_UNCOMMITTED_ROWS,
-                    uncommittedChannels.stream()
-                        .map(SnowflakeStreamingIngestChannelInternal::getFullyQualifiedName)
-                        .collect(Collectors.toList()));
-              }
-            });
+      // here we flush even if removal fails because, for some reason,
+      // the client sequencers did not match.
+      // This does not seem like the desirable behavior.
```

Comment:
I don't think I understand what you mean here, could you elaborate? But I notice a potential wrong result issue from your question.... please take a look if you're interested. PR: https://github.com/snowflakedb/snowflake-ingest-java/pull/163/files

Reply:
@sfc-gh-tzhang The `removeChannelIfSequencersMatch()` removes the channel from the cache if 1) it was there in the first place and 2) if the client seq matches. If all those conditions are satisfied, then it makes sense to call `flush()`, upload and register the created blob. If any of these conditions does not match, and the removal fails (e.g. the client seq is old), then should we flush and upload? or should we just invalidate the channel and move on? 

I am not sure if this scenario is realistic but I just wanted to point this out.

Reply:
Yeah, it makes sense, the validation check is actually in `checkValidation()` at the top of this function. If the channel is not in the channel cache, it must be invalid, you could add a logWarn to verify this logic